src/DiagonalPreconditioner.cpp

```diff
@@ -174,7 +176,7 @@ void DiagonalPreconditioner::SetDefaultScalings() {
         } else {
             DREAM::IO::PrintWarning(
                 "DiagonalPreconditioner: Unrecognized unknown '%s'. Unknown "
-                "will not be preconditioned."
+                "will not be preconditioned.", name
```

Comment:
``name`` -> ``name.c_str()``. This appears to be the reason the LLVM test fails.